### DISCUSSION TITLE
Signing Transaction Using Aptos Wallet Adapter

### DISCUSSION THREAD

**Discussion by abdul-kabugu** [aptos-labs/aptos-developer-discussions#452] [Questions] [answered]
### Discord user ID

kabugu

### Describe your question in detail.

I'm trying to use the AptosWallet adapter to sign a MoveFlow transaction, but I'm encountering an issue. Here's what my payload looks like:
```
const payload = sdk.stream.create({
  recipientAddr: '0x09d29f0ec5b03fc73b59e35deb80356cde17b13b8db94ded34fc8130dc1da1d9',
  depositAmount: 1,
  startTime: start_time,
  stopTime: stop_time,
  interval: 1000,
  // Optional params:
  // coinType: AptosCoin,
  // name: '1',
  // remark: '1',
  // canPause: true,
  // closeable: true,
  // recipientModifiable: true
});

```
Here's my code for signing the transaction:
```
const handleCreateStream = async () => {
  const txid = await signAndSubmitTransaction(payload);
  console.log("tx id", txid);
}
```
However, I keep getting an error saying that a property of data is missing. Could you please help me identify what's wrong with the payload or the signing process?

Thanks in advance for your help!

### What error, if any, are you getting?

I keep getting an error saying that a property of data is missing.

### What have you tried or looked at?  Or how can we reproduce the error?

```
const payload = sdk.stream.create({
  recipientAddr: '0x09d29f0ec5b03fc73b59e35deb80356cde17b13b8db94ded34fc8130dc1da1d9',
  depositAmount: 1,
  startTime: start_time,
  stopTime: stop_time,
  interval: 1000,
  // Optional params:
  // coinType: AptosCoin,
  // name: '1',
  // remark: '1',
  // canPause: true,
  // closeable: true,
  // recipientModifiable: true
});
```

Here's my code for signing the transaction:
```
const handleCreateStream = async () => {
  const txid = await signAndSubmitTransaction(payload);
  console.log("tx id", txid);
}

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

TypeScript SDK

### Describe your environment or tooling in detail

_No response_

**Comment by ajaythxkur** [comment]
@abdul-kabugu its probably looking for something like this
`const response = await signAndSubmitTransaction({
                sender: account.address,
                data: {
                    function: `module_address::module_name::function_name}`,
                    typeArguments:[],
                    functionArguments:[],
                }
            }); `

**Comment by 0xmaayan** [comment, marked as answer]
The `signAndSubmitTransaction` method expects a [InputTransactionData](https://github.com/aptos-labs/aptos-wallet-adapter/blob/main/packages/wallet-adapter-core/src/LegacyWalletPlugins/types.ts#L167) object type. Here is an [example](https://github.com/aptos-labs/aptos-wallet-adapter/blob/main/apps/nextjs-example/src/components/transactionFlows/SingleSigner.tsx#L49-L59) on how to use the method to sign and submit a transaction through the wallet adapter
